### PR TITLE
Sqlx 0.7 update

### PR DESCRIPTION
**Pull request by auipga**
[Changelog of sqlx]( https://github.com/launchbadge/sqlx/blob/main/CHANGELOG.md#breaking)
  **TL;DR:**
  - MSSQL feature gone -> https://github.com/SergioBenitez/Rocket/pull/2604/commits/66385706602b160297be5da70f979f04b6e61224
  - offline feature gone -> https://github.com/SergioBenitez/Rocket/pull/2604/commits/076343d06cdde1235af4fae797a7d091ec4ce9e1, https://github.com/SergioBenitez/Rocket/pull/2604/commits/cafcbbf0f5c8f9e7aed899dd715a2d1364bd55c8
  - another dereference needed for `mut db: Connection<Logs>` (🙏🏻[reddit](https://www.reddit.com/r/rust/comments/14pw35f/comment/jqmv7gy/)) -> https://github.com/SergioBenitez/Rocket/pull/2604/commits/5e15bc7b299e6baa1ae188186361c107245fc02f
  - [opt is consuming self](https://github.com/launchbadge/sqlx/commit/b72a52b0666f69965480c86137e5840d0b361d0a#diff-eab8ee12bc9e34ac092d051bbd03835c90f323cfc96d67956ca7b9be02db70cdR180), changed somewhere between 0.6.3 and 0.7.0-alpha.1 -> https://github.com/SergioBenitez/Rocket/pull/2604/commits/acf3b73a8ee92e85e47010e4e3d60fa3737c1e4d

Fixes: "use two different sqlx versions?" when using shuttle.rs

- [X] All tests are passing

---

I'm new to Rust. This is my first contribution to the Rust community. Please tell me, if you would have done this better. I'm willing to learn.

### CODE REVIEW

**Comment by SergioBenitez** [comment]
This appears to be doing something similar to what #2597 does. Is there a reason to prefer one or the other? Does #2597 actually subsume the changes in this PR?

**Comment by auipga** [comment]
Oh man, same work done twice 🤦🏻‍♂️ At least I learned something by doing it myself.

The only differences to https://github.com/SergioBenitez/Rocket/pull/2597 that I spotted are options for `tls`:
- https://github.com/SergioBenitez/Rocket/pull/2597/commits/f96d120eb544c92a4aaaf9f97c99100b2784bf07#diff-b07af3edcc9e0cbef28b433c304b22197ae60b092d0b4496c3be9cbb9715cc57R40-R42
- https://github.com/SergioBenitez/Rocket/commit/6909a2c88d63b5d835d25cc9892d57e287ca2744#diff-2704e80751ab4993f7c6bc12f6d7ab9a69abeddc182fa5cd47867a582f8ade15R256-R284

To your questions:
- both PRs will do the job. Though mine has no reformats.
- yes, plus the added tls fields (I'd think this shouldn't be included in an regular update)

**Comment by SergioBenitez** [comment]
I was about to merge (a version of) this when I noticed spurious failures caused by a change I made to the sqlx example:

```diff
 #[post("/", data = "<post>")]
-async fn create(mut db: Connection<Db>, post: Json<Post>) -> Result<Created<Json<Post>>> {
-    // There is no support for `RETURNING`.
-    sqlx::query!("INSERT INTO posts (title, text) VALUES (?, ?)", post.title, post.text)
-        .execute(&mut *db)
-        .await?;
+async fn create(mut db: Connection<Db>, mut post: Json<Post>) -> Result<Created<Json<Post>>> {
+    let query = sqlx::query! {
+        "INSERT INTO posts (title, text) VALUES (?, ?) RETURNING id",
+        post.title, post.text
+    };

+    post.id = Some(query.fetch_one(&mut **db).await?.id);
     Ok(Created::new("/").body(post))
 }
```

This is something I had been wanting the example to do for some time but we lacked an updated version of `sqlite`, which we now have. Unfortunately, this causes spurious test failures because an insertion is at times not reflected:

```
DELETE /sqlx:
   >> Matched: (destroy) DELETE /sqlx
   >> Outcome: Success
POST /sqlx application/json:
   >> Matched: (create) POST /sqlx
   >> Outcome: Success
GET /sqlx:
   >> Matched: (list) GET /sqlx
   >> Outcome: Success
POST /sqlx application/json:
   >> Matched: (create) POST /sqlx
   >> Outcome: Success
GET /sqlx:
   >> Matched: (list) GET /sqlx
   >> Outcome: Success
thread 'tests::test_sqlx' panicked at databases/src/tests.rs:38:9:
assertion `left == right` failed
  left: 1
 right: 2
```

The `sqlx` version of the example is the only one that fails even though I've updated the other examples to use `RETURNING` as well. We can't merge this change until we understand why this is happening and get a fix in wherever appropriate. (P.S: I considered whether the defaults for sqlite connections had changed in `sqlx`, e.g, whether WAL is still default, but it seems they have not.)

I've pushed the was-ready-to-merge `sqlx-0.7-update` for your investigation.

**Comment by SergioBenitez** [comment]
Looks like the issue I outline above is https://github.com/launchbadge/sqlx/issues/2543. I'm rather surprised that this seemingly very serious issue has been completely ignored for over two months, or longer (https://github.com/launchbadge/sqlx/issues/1648).

**Comment by SergioBenitez** [comment]
Worked around the issue. Merged in aa7805a5f838daa906b64eae7104584414a63e81. Thank you!
